flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManager.java

```diff
@@ -284,7 +284,7 @@ public void close() throws Exception {
     @Override
     public void clearResourceRequirements(JobID jobId) {
         jobMasterTargetAddresses.remove(jobId);
-        taskManagerTracker.clearPendingAllocationsOfJob(jobId);
+        taskManagerTracker.clearPendingAllocationsOfJob(jobId, resourceAllocator.isSupported());
```

Comment:
We should not pass `resourceAllocator.isSupported()` into `taskManagerTracker`. Instead, we should not call `taskManagerTracker.clearPendingAllocationsOfJob()` if the allocator is not supported.

Reply:
Same for `replaceAllPendingAllocations`.